wipp-backend-argo-workflows/src/main/java/gov/nist/itl/ssd/wipp/backend/argo/workflows/workflow/WorkflowExitController.java

```diff
@@ -84,12 +88,32 @@
         }
 
         Workflow workflow = wippWorkflow.get();
+        ArgoWorkflow argoWorkflow = argoWorkflowRepository.findByName(workflow.getGeneratedName());
 
         // Check validity of status
         WorkflowStatus wfStatus;
-        status = status.toUpperCase();
+        Map<String,String> jobsStatus = new HashMap<>();
+
         try {
-            wfStatus = WorkflowStatus.valueOf(status);
+            wfStatus = WorkflowStatus.valueOf(status.toUpperCase());
+
+            try {
+                JSONObject jsonWorkflow = new JSONObject(argoWorkflow.getWorkflow());
+                JSONObject jsonStatus = jsonWorkflow.getJSONObject("status");
+                JSONObject jsonNodes = jsonStatus.getJSONObject("nodes");
+                Iterator<String> keys = jsonNodes.keys();
+                while (keys.hasNext()){
+                    String key = keys.next();
+                    JSONObject job = jsonNodes.getJSONObject(key);
+
+                    // check that the job is neither the DAG nor the exit handler
+                    if (job.getString("type").equals("Pod") && !job.getString("templateName").equals("exit-handler")){
```

Comment:
Maybe we could map the object into actual java classes?
We could generate them using [Swagger Codegen](https://github.com/swagger-api/swagger-codegen#os-x-users) and the [Argo OpenAPI spec](https://raw.githubusercontent.com/argoproj/argo/v2.4.3/api/openapi-spec/swagger.json)